src/main/java/com/hillel/items_exchange/controller/UserController.java

```diff
@@ -0,0 +1,46 @@
+package com.hillel.items_exchange.controller;
+
+import com.hillel.items_exchange.dto.UserDto;
+import com.hillel.items_exchange.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.Principal;
+import java.util.Locale;
+
+@RestController
+@RequestMapping("/user")
+@RequiredArgsConstructor
+@Slf4j
+public class UserController {
+
+    private final MessageSource messageSource;
+    private final UserService userService;
+
+    @GetMapping("/info/{id}")
+    public @ResponseBody
+    ResponseEntity<UserDto> getUserInfo(@Validated @PathVariable("id") long id, Principal principal) {
```

Comment:
Why do you use Validated annotation here?

Reply:
Earlier, you asked to perform id validation with Spring. That's what I did. My choice isn't right?